Sklep/Utils/BarcodeScanner.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AForge.Video;
+using AForge.Imaging;
+using AForge.Math;
+using AForge.Video.DirectShow;
+using ZXing;
+using AForge.Imaging.Filters;
+using System.Drawing;
+using System.Windows.Forms;
+namespace Sklep.Utils
+{
+    public class BarcodeScanner
+    {
+        public event EventHandler<string> CodeScanned; // event
+
+        FilterInfoCollection filterInfoCollection;
+        VideoCaptureDevice videoCaptureDevice;
+        ContrastCorrection filter;
+        BarcodeReader reader;
+        public PictureBox pictureBox = null;
+
+        public void startScanning()
```

Comment:
Tutaj widzę jeden duży problem - startScanning() robi więcej rzeczy niż rozpoczęcie skanowania. Rejestrujesz chociażby metodę VideoCaptureDevice_NewFrame, co należy robić jedynie raz. Obecnie wywołanie startScanning() stopScanning() i startScanning() spowoduje dwukrotne dodanie tego eventu.
Albo w stopScanning trzeba odrejestrować event VideoCaptureDevice_NewFrame, albo przenieść to do konstruktora, a w startScanning() wywołać jedynie videoCaptureDevice.Start();